lib/data_feeds/perse_api.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module DataFeeds
+  module PerseApi
+    def self.connection
+      Faraday.new(ENV.fetch('PERSE_API_URL'), headers: { content_type: 'application/json',
+                                                         'api_key' => ENV.fetch('PERSE_API_KEY') }) do |f|
+        f.response :json
+        f.response :raise_error
+        # f.response :logger
+      end
+    end
+
+    def self.meter_history_realtime_data(mpan, from_date)
+      params = { MPAN: mpan, fromDate: from_date.respond_to?(:strftime) ? from_date.strftime('%Y-%m-%d') : from_date }
+      connection.get('/meterhistory/v2/realtime-data', params).body
+    rescue Faraday::Error => e
+      EnergySparks::Log.exception(e, mpan:, from_date:, status: e&.response_status, body: e&.response_body)
+      {}
+    end
+
+    def self.meter_history_readings(mpan, from_date)
+      meter_history_realtime_data(mpan, from_date)['data']
+        &.select { |data| data_ok(data) }
+        &.map { |data| [data['Date'], (1..48).map { |i| data["P#{i}"].to_f }] } || []
```

Comment:
Are there ever any missing or blank half-hourly readings? We'll want missing/blank to be left as nil so we end up storing what was returned via the API. Otherwise they get set to zero here.